src/App.js

```diff
@@ -2,66 +2,59 @@ import React, { Fragment, useEffect, useState } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
-// ### Side Effects ###
 
-// Side effects are produced by impure functions. They result as a consequence of the
-// actions of impure functions. It is an effect outside of the function which isn't 
-// captured directly in its return value. In this implementation, the side effects is
-// includes:
-
-// ### 1 ###
-// Default state variable stored in localStorage using conditional(ternary) operator
-// to handle string and null primitive data type values.
-
-// ### 2 ###
-// Changing of the DOM by the useState() hook in the custom hook.
-
-// ### 3 ###
-// The filter() method creates a new array of elements that passed the condition provided in it.
-// The existing array is traversed and only the todo element whose id matches the argument accepted,
-// is filtered out. This new array is then passed as an argument for the setter function.
-
-
-function useSemiPersistentState() {
-  
-  const persistentList = localStorage.getItem("savedTodoList")
-  ? JSON.parse(localStorage.getItem("savedTodoList"))
-  : [];  // * 1 * //
+function App() {
 
-  const [todoList, setTodoList] = useState(persistentList);
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList))
-  }, [todoList]); 
-
-  return [todoList, setTodoList];
+    new Promise((resolve, reject) =>
+      setTimeout(
+        () => 
+          resolve({
+            data : {
+              todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
+            },
+          }), 
+        2000
+      )
+      ).then((result) => {
+        setTodoList(result.data.todoList);
+        setIsLoading(false);
+    });
+  }, []);
 
-};
-
-
-function App() {
-
-  const [todoList, setTodoList] = useSemiPersistentState();
+  useEffect(() => {
+    if(isLoading){
```

Comment:
I'll slack you a recording, but this hook isn't working as intended - this conditional statement means we'll only run this hook while the app is loading - but that's not the expected result! I attached a screenshot of the instruction I'm talking about.
<img width="1070" alt="Screenshot 2023-03-07 at 9 01 20 AM" src="https://user-images.githubusercontent.com/5691923/223477725-cf8bc813-cfca-4ce3-9a9e-4acea9db2424.png">

Reply:
Thanks Mark.

I will check slack for the details.

I appreciate.

Reply:
Hi Mark!

I just fixed the bug and changes have been pushed please.

Thanks so much.

I appreciate.